src/ILCompiler.Build.Tasks/src/ComputeManagedAssemblies.cs

```diff
@@ -0,0 +1,175 @@
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+
+
+namespace Build.Tasks
+{
+    public class ComputeManagedAssemblies : Task
+    {
+        [Required]
+        public ITaskItem[] Assemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The CoreRT-specific System.Private.* assemblies that must be used instead of the netcoreapp2.0 versions.
+        /// </summary>
+        [Required]
+        public ITaskItem[] SdkAssemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The set of AOT-specific framework assemblies we currently need to use which will replace the same-named ones
+        /// in the app's closure.
+        /// </summary>
+        [Required]
+        public ITaskItem[] FrameworkAssemblies
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The native apphost (whose name ends up colliding with the CoreRT output binary) 
+        /// </summary>
+        [Required]
+        public string DotNetAppHostExecutableName
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The CoreCLR dotnet host fixer library that can be skipped during publish
+        /// </summary>
+        [Required]
+        public string DotNetHostFxrLibraryName
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The CoreCLR dotnet host policy library that can be skipped during publish
+        /// </summary>
+        [Required]
+        public string DotNetHostPolicyLibraryName
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] ManagedAssemblies
+        {
+            get;
+            set;
+        }
+
+        [Output]
+        public ITaskItem[] AssembliesToSkipPublish
+        {
+            get;
+            set;
+        }
+
+        public override bool Execute()
+        {
+            var list = new List<ITaskItem>();
+            var assembliesToSkipPublish = new List<ITaskItem>();
+
+            var coreRTFrameworkAssembliesToUse = new HashSet<string>();
+
+            foreach (ITaskItem taskItem in SdkAssemblies)
+            {
+                coreRTFrameworkAssembliesToUse.Add(Path.GetFileName(taskItem.ItemSpec));
+            }
+
+            foreach (ITaskItem taskItem in FrameworkAssemblies)
+            {
+                coreRTFrameworkAssembliesToUse.Add(Path.GetFileName(taskItem.ItemSpec));
+            }
+
+            foreach (ITaskItem taskItem in Assemblies)
+            {
+                // In the case of disk-based assemblies, this holds the file path
+                string itemSpec = taskItem.ItemSpec;
+
+                // Skip crossgen images
```

Comment:
There should not be any `*.ni.dll` images anymore. Do you see any?

Reply:
There are none and none are being passed as inputs. Are we still supporting them?

Reply:
We are supporting them, but we are not using .ni.dll suffix for them.